core/framework/condition_collection.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import "github.com/kiegroup/kogito-cloud-operator/api"
+
+// ConditionCollection ...
+type ConditionCollection interface {
+	AddCondition(condition api.ConditionInterface) []api.ConditionInterface
+}
+
+type conditionCollection struct {
+	conditions []api.ConditionInterface
+}
+
+// NewConditionCollection ...
+func NewConditionCollection(conditions []api.ConditionInterface) ConditionCollection {
+	return &conditionCollection{
+		conditions: conditions,
+	}
+}
+
+func (c *conditionCollection) AddCondition(newCondition api.ConditionInterface) []api.ConditionInterface {
+	for i, condition := range c.conditions {
+		if condition != nil && condition.GetType() == newCondition.GetType() {
+			c.removeOldCondition(i)
```

Comment:
shouldn't be better keep, like, the latest X conditions here?

Reply:
maybe set a limit and use FILO?

Reply:
Previous it was implemented in that manner only. But in jira the expectation is to behave like a collection of the latest status of each different type.

Reply:
Each condition should be a snapshot of the latest status. It should be have like a key-value map, not like a history

Reply:
Please see: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L30

We should use this behavior and remove the `buffer` thing.

Reply:
We have access to this function, it's just a matter of updating our Status conditions.

Reply:
I see, ok, is this question resolved?

Reply:
As discussed, I had implemented condition as a snapshot of the latest status and use `k8s.io/apimachinery/pkg/api/meta` Condition object